genkit-tools/cli/src/mcp/trace.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RuntimeManager } from '@genkit-ai/tools-common/manager';
+import { McpServer } from '@modelcontextprotocol/sdk/server/mcp';
+import z from 'zod';
+
+export function defineTraceTools(server: McpServer, manager: RuntimeManager) {
+  server.registerTool(
+    'get_trace',
+    {
+      title: 'Get Trace',
+      description: 'Returns the trace details',
+      inputSchema: {
+        traceId: z
+          .string()
+          .describe(
+            'trace id (typically returned after running a flow or other actions)'
+          ),
+      },
+    },
+    async ({ traceId }) => {
+      try {
+        const response = await manager.getTrace({ traceId });
+        return {
+          content: [
+            { type: 'text', text: JSON.stringify(response, undefined, 2) },
+          ],
```

Comment:
This is probably fine as a start, but I'd recommend a simplified text-friendly trace format when you have some time.

Reply:
Agreed... added a todo for now